R/Connection.R

```diff
@@ -712,6 +712,56 @@ setMethod(
     invisible(TRUE)
   })
 
+#' List Locations of ODBC Configuration Files
+#'
+#' @description
+#' On MacOS and Linux, odbc uses the unixODBC driver manager to manage
+#' information about driver and data sources. This helper returns the filepaths
+#' where the driver manager will look for that information.
+#'
+#' @details
+#' This function is a wrapper around the command line call `odbcinst -j`.
+#'
+#' Windows now uses the ODBC Data Source Administrator application to manage
+#' drivers and thus does not use `.ini` files; this function will return a
+#' 0-row data frame on Windows.
+#'
+#' @seealso
+#' The [odbcListDrivers()] and [odbcListDataSources()] helpers return
+#' information on the contents of `odbcinst.ini` and `odbc.ini` files,
+#' respectively.
+#'
+#' Learn more about unixODBC and the `odbcinst` utility
+#' [here](https://www.unixodbc.org/odbcinst.html).
+#'
+#' @examplesIf FALSE
+#' configs <- odbcListConfig()
+#'
+#' file.edit(configs$location[1])
```

Comment:
My main question is if this should just be a named character vector with names "drivers", "system_dsn", "user_dsn", or similar.

Reply:
You got it—[680b502](https://github.com/r-dbi/odbc/pull/680/commits/680b50228c201b284d3e93be45b0b2cdcd09cfbb).

Reply:
Heard. Had thought about that but I wasn't sure whether folks might end up with multiple user `odbc.ini` files with different levels of permissions/scope. Trying to find some documentation that explicitly states that there is only ever one user-level `odbc.ini` file but I'll run with it if we're confident that that's the case.

Reply:
Okay, from the source, looks like there can only ever be one active user-level `odbc.ini`. Glad to make this change if you feel we ought to!

Reply:
Yeah, lets do it.

Reply:
[6689544](https://github.com/r-dbi/odbc/pull/680/commits/6689544ebfb54fe6d8ee6034030e377ba583b85c)!